File: hushh-adk-agents/._kushal_profile_data/Interview-Prep/Problems_By_Pattern/Two_Pointers/Container_Most_Water/container_with_most_water.ts
function maxArea(height: number[]): number {
    let left = 0;
    let right = height.length - 1;
    let maxWater = 0;
    
    while (left < right) {
        const h = Math.min(height[left], height[right]);
        const w = right - left;
        const area = h * w;
        
        maxWater = Math.max(maxWater, area);
        
        if (height[left] < height[right]) {
            left++;
        } else {
            right--;
        }
    }
    
    return maxWater;
};

File: hushh-adk-agents/._kushal_profile_data/Interview-Prep/Problems_By_Pattern/Two_Pointers/Container_Most_Water/container_with_most_water.cs
public class Solution {
    public int MaxArea(int[] height) {
        int left = 0;
        int right = height.Length - 1;
        int maxWater = 0;
        
        while (left < right) {
            // Calculate area: min(height) * width
            int h = Math.Min(height[left], height[right]);
            int w = right - left;
            int area = h * w;
            
            maxWater = Math.Max(maxWater, area);
            
            // Move the shorter line inward
            if (height[left] < height[right]) {
                left++;
            } else {
                right--;
            }
        }
        
        return maxWater;
    }
}
